script.js

```diff
@@ -0,0 +1,86 @@
+var box = $("td"); //Array used to check for the winning patterns.
+var playerTurn = 0; //Variable to switch player turns after each click.
+var playerChoiceAtEnd; //Variable which will decide if the player wants to play again or not.
+
+
+//Function which is responsible for colouring the boxes according to which players turn it is.
+function clickedBox(){
+    $("td").click(function(){
+        if(this.classList != "X" && this.classList != "O"){
+            if(playerTurn == 0){
+                $(".playerTurn").html("It is O's turn");
+                $(this).html("X").addClass("X");
+                playerTurn = 1;
+            } else{
+                $(".playerTurn").html("It is X's turn");
+                $(this).html("O").addClass("O");
+                playerTurn = 0;
+            }
+        }
+    });
+}
+
+//Function to reset the game by clearing the classes.
+function reset(){
+    $("td").each(function(){
+        if($(this).hasClass("X")){
```

Comment:
If you're just clearing the board, you don't need to check which have a class and which don't - you can just take it off all of them.